src/components/brightness/BrightnessController.h

```diff
@@ -20,6 +23,25 @@ namespace Pinetime {
 
     private:
       Levels level = Levels::High;
+      static constexpr uint8_t UNSET = UINT8_MAX;
```

Comment:
Since all those constants are implementation details (nothing except the code in BrightnessController.cpp use them) , they can be defined in `BrightnessController.cpp`. The includes for nrf_ppi.h and nrfx_gpiote.h will also move to the cpp file. 

This won't change anything except that it cleans the header (interface) of the class.

Reply:
I agree that UNSET and its definition can definitely move in, so can the the gpiote header.

Do you essentially mean that all private constants should move into the .cpp file (as they are private so by definition they can only be used by the corresponding class).
If so, this doesn't follow how a lot of InfiniTime is currently written.

I view constants declared in the private section of the header as constants that one may want to review or change in the future. IMO things like the PWM frequency or brightness steps available are things that should be explicitly configurable (while the exact value of UNSET does not need to be, for example).

Where do you draw the line between a private constant in the header and a constant defined in the implementation?

Reply:
> If so, this doesn't follow how a lot of InfiniTime is currently written.

Yes, you're right! It's just that I learn everyday, and I noticed that some project put their private (aka implementation detail) into their .cpp file so that they are not visible at all to the 'users' of the class. This allows to keep the header of the class cleaner (you only expose what is interesting to the users of the class) and it also remove the coupling between classes by hiding dependence (like this include to the NRF SDK in this case). 

This might not be a good idea in all cases and for all variables though... it's just a feeling :)

Also, hiding the dependency to the NRF SDK might help when we'll eventually support multiple platform, but this is not the case right now.

Reply:
OK, so my take is that tunables for a class e.g BLE address for a characteristic, screen width / height, filesystem size should live in the header, as changing these should not require changing the implementation (.cpp). These are all examples of current private fields. Would these in your opinion better belong in the implementation? I'm trying to feel out your exact views here so I know what to do in the future :)

Since they are private, users of the class are unbothered anyway (no access), but being able to glance at the filesystem size for example may well be useful to a programmer using the class.

Regarding coupling, I've got a WIP/RFC resource map prototype
```c++
#pragma once
#include <cstdint>
#include "nrf_ppi.h"

namespace Pinetime {
  namespace ResourceMap {
    // PPI
    // 20 configurable channels, 5 used
    // Channels 3,6-19 are unused
    // Channels 20 through 31 (inclusive) have fixed function (mostly BLE radio related)
    // Refer to https://infocenter.nordicsemi.com/topic/com.nordic.infocenter.nrf52832.ps.v1.1/ppi.html

    static constexpr nrf_ppi_channel_t SpiErratumWorkaround = NRF_PPI_CHANNEL0;
    static constexpr nrf_ppi_channel_t BacklightOn = NRF_PPI_CHANNEL1;
    static constexpr nrf_ppi_channel_t BacklightOff = NRF_PPI_CHANNEL2;

    // FOR INFORMATION ONLY: defined in ble_phy.c
    static constexpr nrf_ppi_channel_t NimbleResponseTimerDisable = NRF_PPI_CHANNEL4;
    static constexpr nrf_ppi_channel_t NimbleResponseTimerWait = NRF_PPI_CHANNEL5;
    // FOR INFORMATION ONLY

    // RTC
    // 3 RTCs, all used

    // FOR INFORMATION ONLY: hardcoded across nimble (ble_phy.c, hal_timer.c)
    static constexpr uint32_t NimbleRTC = NRF_RTC0_BASE;
    // FOR INFORMATION ONLY

    // FOR INFORMATION ONLY: defined in portmacro_cmsis.h
    static constexpr uint32_t FreeRTOSRTC = NRF_RTC1_BASE;
    // FOR INFORMATION ONLY

    static constexpr uint32_t BacklightRTC = NRF_RTC2_BASE;

    // TIMER
    // 5 timers, 2 used
    // Timers 1,3,4 are unused

    // FOR INFORMATION ONLY: hardcoded across nimble (ble_phy.c)
    static constexpr uint32_t NimbleTimer = NRF_TIMER0_BASE;
    // FOR INFORMATION ONLY

    // Timer 2 CC0 is used to store the firmware version
    // Other than that the timer is unused
    static constexpr uint32_t FirmwareVersionStorageTimer = NRF_TIMER2_BASE;
  }
}
```
Which would clean that up. I think a central definition of peripherals used does make sense. Not sure if this form is the ideal one though.

Reply:
> Would these in your opinion better belong in the implementation? I'm trying to feel out your exact views here so I know what to do in the future :)

I'm currently working (on my day job) on big projects with a lot of internal and external dependencies. In this context, I noticed that it's far easier to integrate a class/library that exposes the least amount of types and other dependencies. So yeah, I think that everything that's not needed by the code that uses the class should be hidden in the .cpp file. 
You're right that this could serve as documentation for the developer though so... I guess the idea is not mature enough for me to write an actual coding guideline.

Your resource map looks similar to the current `PinMap` (https://github.com/InfiniTimeOrg/InfiniTime/blob/main/src/drivers/PinMap.h) we have in InfiniTime : I'm not a huge fan on this, since this effectively creates global variables, but I guess this is a effective way to document the resource usage so... let's use it :-)

Anyway, this discussion it probably out of the scope of this PR, let's move the variables to .cpp if we ever feel the need in the future :)